.travis.yml

```diff
@@ -1,7 +1,33 @@
-sudo: required
-services: docker
-env:
-  - CC=gcc CXX=g++
-  - CC=clang CXX=clang++
+language: cpp
+
+matrix:
+  include:
+    - os: linux
+      dist: trusty
+      sudo: required
+      services: docker
+      compiler: gcc
+    - os: linux
+      dist: trusty
+      sudo: required
+      services: docker
+      compiler: clang
+    - os: osx
+      compiler: clang
+
+install:
+  - if [ "${TRAVIS_OS_NAME}" = "osx" ]; then QT_CI_PACKAGES=qt.595.clang_64,qt.595.qtwebengine PATH=$PATH:$PWD/tools ./tools/qt-5.9.5-osx; fi
```

Comment:
Does the yaml allow some of these tests to be collapsed? Ignore if not.

Reply:
yes you can use a yaml pipe (|) on a sequence (-), i.e "- |", but it seems to be very seldom used in .travis.yml files.  A more standard approach is to move all but trivial things into scripts in your repo, I have taken this approach as recommended by https://docs.travis-ci.com/user/customizing-the-build#Implementing-Complex-Build-Steps